nuxeo-features/rest-api/nuxeo-rest-api-server/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/BatchUploadObject.java

```diff
@@ -104,6 +111,75 @@
 
     public static final String UPLOAD_TYPE_CHUNKED = "chunked";
 
+    private BatchManager batchManager;
+
+    public static class Constants {
+        public static final String MD5 = "md5";
+        public static final String MIMETYPE = "mimeType";
+        public static final String NAME = "name";
+        public static final String FILE_SIZE = "fileSize";
+        public static final String KEY = "key";
+    }
+
+    @Override protected void initialize(Object... args) {
+        super.initialize(args);
+        batchManager = Framework.getService(BatchManager.class);
+    }
+
+    @GET
+    @Path("handlers")
+    public Response providers() throws IOException {
+        Set<String> supportedHandlers = batchManager.getSupportedHandlers();
+        Map<String, Object> result = new HashMap<>();
+
+        result.put("handlers",
+                supportedHandlers
+                                 .stream()
+                                 .map(entry -> Collections.singletonMap("name",
+                                         entry))
+                                 .collect(Collectors.toList())
+        );
+
+        return buildResponse(Status.OK, result);
+    }
+
+    @GET
+    @Path("handlers/{handlerId}")
+    public Response getProviderInfo(@PathParam("handlerId") String handlerId) throws IOException {
+        BatchHandler handler = batchManager.getHandlerByName(handlerId);
+
+        if (handler == null) {
+            return Response.status(Status.NOT_FOUND).build();
+        }
+
+        Map<String, Object> result = Collections.singletonMap("name", handler.getName());
+
+        return buildResponse(Status.OK, result);
+    }
+
+    @POST
+    @Path("new/{providerId}")
+    public Response createNewBatch(@Context UriInfo uriInfo, @PathParam("providerId") String providerId) throws IOException {
+        if (StringUtils.isEmpty(providerId)) {
+            String batchId = batchManager.initBatch();
+            return Response.seeOther(
+                    UriBuilder
+                        .fromResource(BatchUploadObject.class)
+                        .path("/{batchId}")
+                        .build(batchId)
+                    ).build();
```

Comment:
Any reason for using the POST/redirect/GET pattern here? Can't you simply do `return getBatchInfo(batchId);`?

Reply:
Re-use the `GET /upload/{batchId}/info` Endpoint.

Reply:
Yes, I understand this. The question is: why do you explicitly want to do a *redirect + GET* after the POST instead of just returning the result of the method invoked by the `GET /upload/{batchId}/info`?

By using `seeOther` you are sending a 303 to the client which will perform an extra query to the `GET` endpoint.
Is that intended? If so what's the benefit?

Reply:
The point is applying HATEOAS best-practices. If we have an endpoint that already provides the information we want to return, why not use it?
Regarding the HTTP Status Code, it should be 302, not 303. I don't know why Jersey uses 303 for Redirects.

Reply:
OK, I understand the concept of reusing an endpoint.
But what's the true benefit of applying this concept VS the downside of making the client perform an extra HTTP request whereas the server could directly call the code executed by the GET endpoint: `getBatchInfo`?